server/src/main/java/org/jboss/as/server/moduleservice/ExternalModuleSpecService.java

```diff
@@ -78,13 +130,25 @@ public synchronized void start(StartContext context) throws StartException {
 
 
         ServiceModuleLoader.installModuleResolvedService(context.getChildTarget(), moduleIdentifier);
+    }
 
+    private long fileSize(Path p) {
+        try {
+            return Files.size(p);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
     }
 
     @Override
     public synchronized void stop(StopContext context) {
-        VFSUtils.safeClose(jarFile);
-        jarFile = null;
+        for (JarFile jarFile : jarFiles) {
+            log.debugf("Closing %s jar file which was added as resource root for %s module identifier", jarFile.toString(), moduleIdentifier.getName());
+            VFSUtils.safeClose(jarFile);
+            log.debugf("Closed %s jar file which was added as resource root for %s module identifier", jarFile.toString(), moduleIdentifier.getName());
```

Comment:
These debug statements don't add much, as jarFile.toString() is just getClass().getName() + "@" + Integer.toHexString(hashCode())

At most one per file should be sufficient.

Reply:
Ups, sorry, some scrap left from the development